sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/ca/DefaultCAClient.java

```diff
@@ -0,0 +1,159 @@
+package com.mesosphere.sdk.dcos.ca;
+
+import com.mesosphere.sdk.dcos.CertificateAuthorityClient;
+import com.mesosphere.sdk.dcos.DcosConstants;
+import com.mesosphere.sdk.dcos.http.URLHelper;
+import org.apache.http.HttpResponse;
+import org.apache.http.StatusLine;
+import org.apache.http.client.fluent.ContentResponseHandler;
+import org.apache.http.client.fluent.Executor;
+import org.apache.http.client.fluent.Request;
+import org.apache.http.client.fluent.Response;
+import org.apache.http.entity.ContentType;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayInputStream;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+/**
+ * A default implementation of {@link CertificateAuthorityClient}.
+ */
+public class DefaultCAClient implements CertificateAuthorityClient {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private URL baseURL;
+    private Executor httpExecutor;
+    private CertificateFactory certificateFactory;
+
+    public DefaultCAClient(URL baseURL, Executor executor) {
+        this.baseURL = baseURL;
+        this.httpExecutor = executor;
+
+        try {
+            this.certificateFactory = CertificateFactory.getInstance("X.509");
+        } catch (CertificateException e) {
+            logger.error("Failed to create certificate factory", e);
+        }
+    }
+
+    public DefaultCAClient(Executor executor) {
+        this(URLHelper.fromUnchecked(DcosConstants.CA_BASE_URI), executor);
+    }
+
+    @Override
+    public X509Certificate sign(byte[] csr) throws Exception {
```

Comment:
There is also possibility of `java.security.cert.CertificateException` when PEM string is decoded. I had originally 2 distinct exceptions and moved it to common single Exception.